grails-app/controllers/au/org/ala/specieslist/SpeciesListItemController.groovy

```diff
@@ -42,6 +47,12 @@ class SpeciesListItemController {
                     flash.message = "${message(code: 'default.not.found.message', args: [message(code: 'speciesList.label', default: 'Species List'), requestParams.id])}"
                     redirect(controller: "public", action: "speciesLists")
                 } else {
+                    if (!securityUtil.checkListAccess(params.id)) {
```

Comment:
Cannot this be done in a less coupled / perocedural way? Security should be a cross cutting concern.